src/mono/mono/mini/aot-compiler.c

```diff
@@ -13215,6 +13220,124 @@ add_profile_instances (MonoAotCompile *acfg, ProfileData *data)
 	printf ("Added %d methods from profile.\n", count);
 }
 
+static int
+add_single_mibc_profile_method (MonoAotCompile *acfg, MonoMethod *method)
+{
+	if (!method->is_inflated || mono_method_is_generic_sharable_full (method, FALSE, FALSE, FALSE)) {
+		if (!acfg->aot_opts.profile_only)
+			return 0;
+		if (m_class_get_image (method->klass) != acfg->image)
+			return 0;
+		add_profile_method (acfg, method);
+		return 1;
+	}
+
+	if (acfg->aot_opts.dedup_include) {
+		add_profile_method (acfg, method);
+		return 1;
+	}
+
+	MonoGenericContext *ctx = mono_method_get_context (method);
+	if ((ctx->class_inst && inst_references_image (ctx->class_inst, acfg->image)) ||
+		(ctx->method_inst && inst_references_image (ctx->method_inst, acfg->image))) {
+		add_profile_method (acfg, method);
+		return 1;
+	}
+
+	if (m_class_get_image (method->klass) == acfg->image &&
+		((ctx->class_inst && is_local_inst (ctx->class_inst, acfg->image)) ||
+		(ctx->method_inst && is_local_inst (ctx->method_inst, acfg->image)))) {
+		add_profile_method (acfg, method);
+		return 1;
+	}
+
+	return 0;
+}
+
+static int
+add_mibc_group_method_methods (MonoAotCompile *acfg, MonoMethod *mibcGroupMethod, MonoImage *image, MonoClass *mibcModuleClass, MonoGenericContext *context)
+{
+	ERROR_DECL (error);
+
+	MonoMethodHeader *mibcGroupMethodHeader = mono_method_get_header_internal (mibcGroupMethod, error);
+	mono_error_assert_ok (error);
+
+	int count = 0;
+	uint8_t *cur = (uint8_t*)mibcGroupMethodHeader->code;
+	uint8_t *end = (uint8_t*)mibcGroupMethodHeader->code + mibcGroupMethodHeader->code_size;
+	while (cur < end) {
+		MonoOpcodeEnum il_op;
+		const unsigned char *opcodeIp = (unsigned char*)cur;
+		const unsigned char *opcodeEnd = (unsigned char*)end;
+		cur += mono_opcode_value_and_size (&opcodeIp, opcodeEnd, &il_op);
+		if (il_op != MONO_CEE_LDTOKEN)
```

Comment:
I wonder if we should follow the mibc spec a little bit more closely in case they use `ldtoken` for other stuff in the future. 

That is: the method body is logically a sequence of groups that start with an `ldtoken <method_token>` and then there are a bunch of other instructions until a `pop`.

So the loop above would do:

1. while (cur < end)
2. assert the current instructoin is `ldtoken <tok>` and `<tok>` is a memberref token
3. *do whatever we need to do to pluck out the method and save it*
4. do { move `cur` to the next instruction } while (cur is not `pop`)
4. move `cur` to the next instruction (ie skip the `pop` too)

Reply:
We will need to account for little and big endian systems here? Whats the best way assert the token is a memberref token then?

Reply:
Looks like we are also hitting MethodSpec's
```
Method 'Assemblies_System.Console;System.Private.CoreLib;_3' (#2f5) (0x06000003)
{
  // Code size        6 (0x6)
  .maxstack  8
  IL_0000:  ldtoken    0x2B000010
  IL_0005:  pop
}
```

Reply:
> are there other acceptable tokens?

ECMA-335 says: 
> A methoddef, methodref or methodspec: pushes a RuntimeMethodHandle

but `methoddef` is only going to be a reference to a method that's defined in the current image (ie in the mibc) which... doesn't make sense for mibc.

btw: we have some helpers for this:

https://github.com/dotnet/runtime/blob/08a7b2382799082eedb94d70fca6c66eb75f2872/src/mono/mono/metadata/verify-internals.h#L17-L18

(I wouldn't use `verify-internals.h`  maybe, but definitely do use `mono_metadata_token_table` and the table definitions instead of 0x2b etc)

Reply:
Done! Thanks